clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -6627,6 +6627,7 @@ void Sema::checkClassLevelDLLAttribute(CXXRecordDecl *Class) {
         auto *Ctor = dyn_cast<CXXConstructorDecl>(MD);
         if ((MD->isMoveAssignmentOperator() ||
              (Ctor && Ctor->isMoveConstructor())) &&
+            getLangOpts().isCompatibleWithMSVC(LangOptions::MSVC2010) &&
```

Comment:
This tripped me up when I first looked at it. I think it would be nice to introduce a  no argument form `bool isCompatibleWithMSVC() const;` and then write the MSVC compatibility bit as: `getLangOpts().isCompatibleWithMSVC() && !getLangOpts().isCompatibleWithMSVC(LangOptions::MSVC2015)`